src/v/transform/rpc/deps.cc

```diff
@@ -146,7 +147,9 @@ class partition_manager_impl final : public partition_manager {
             }
             co_return response;
         }
-        auto stm = partition->transform_offsets_stm();
+        auto stm
+          = partition->raft()->stm_manager()->get<transform_offsets_stm_t>(
+            "distributed_kv_stm");
```

Comment:
Mismatch between the name and the type looks weird and implies that we can't have e.g. more than 1 distributed_kv_stm per partition. Maybe it is better to pass the stm name when registering the factory instead of tying it to the stm type.

Reply:
good point, i am not a fan of those names either, will add ability to pass a name as a part of factory registration interface

Reply:
We should be able to do `partition->raft()->stm_manager()->get<stm_type>()` right? And key everything off the class type?

So the underlying container is `absl::flat_hash_map<std::type_index, state_machine_manager::named_stm>` and the `get` method can create the index using: `std::type_index(typeid(T))`.

This assumes that each stm is it's own type, which I think is fine. For the case of the distributed_kv, I'd expect them to all have different keys and values, otherwise we can add a tag type to distinguish

Reply:
i've ended up with adding a static `name` field to stm this way we do not need to relay ion `typeid` in human readable parts f.e. snapshot map or logs

Reply:
Yeah much better - love it!